cmd/encryption-methods.go

```diff
@@ -194,43 +196,48 @@ func parseSSEKey(sseKey string, keyType sseKeyType) (
 	key string,
 	err *probe.Error,
 ) {
-	if keyType == sseS3 {
-		alias, prefix = splitKey(sseKey)
+	sseKeyBytes := []byte(sseKey)
+
+	seperatorIndex := bytes.LastIndex(sseKeyBytes, []byte("="))
+	if seperatorIndex < 0 {
+		err = errSSEKeyMissing().Trace(sseKey)
 		return
 	}
 
-	var path string
-	alias, path = splitKey(sseKey)
-	splitPath := strings.Split(path, "=")
-	if len(splitPath) == 0 {
+	encodedKey := string(sseKeyBytes[seperatorIndex+1:])
+	if seperatorIndex == len(sseKeyBytes)-1 {
 		err = errSSEKeyMissing().Trace(sseKey)
 		return
 	}
 
-	aliasPlusPrefix := strings.Join(splitPath[:len(splitPath)-1], "=")
-	prefix = strings.Replace(aliasPlusPrefix, alias+"/", "", 1)
-	key = splitPath[len(splitPath)-1]
+	alias, prefix = splitKey(string(sseKeyBytes[:seperatorIndex]))
 
-	if keyType == sseC {
-		keyB, de := base64.RawStdEncoding.DecodeString(key)
-		if de != nil {
-			err = errSSEClientKeyFormat("One of the inserted keys was " + strconv.Itoa(len(key)) + " bytes and did not have valid base64 raw encoding.").Trace(sseKey)
-			return
-		}
-		key = string(keyB)
-		if len(key) != 32 {
-			err = errSSEClientKeyFormat("The plain text key was " + strconv.Itoa(len(key)) + " bytes but should be 32 bytes long").Trace(sseKey)
-			return
-		}
+	if keyType == sseS3 {
+		return
 	}
 
 	if keyType == sseKMS {
-		if !validKMSKeyName(key) {
-			err = errSSEKMSKeyFormat("One of the inserted keys was " + strconv.Itoa(len(key)) + " bytes and did not have a valid KMS key name.").Trace(sseKey)
+		if !validKMSKeyName(encodedKey) {
+			err = errSSEKMSKeyFormat(fmt.Sprintf("Key (%s) is badly formatted.", encodedKey)).Trace(sseKey)
+		}
+		return
+	}
+
+	keyB, de := hex.DecodeString(encodedKey)
+	if de != nil {
+		keyB, de = base64.RawStdEncoding.DecodeString(encodedKey)
```

Comment:
I remember this one, 

base64.RawStdEncoding.DecodeString() will reject the = padding, but base64.StdEncoding.DecodeString() will not reject it.

Also I think there is a possibility that when a user intents to pass base64, it will still be interpreted as hex, for example, both:

	base64.RawStdEncoding.DecodeString("abcd")
	hex.DecodeString("abcd")

will not return an error

I think it is okay to trim all the trailing = in the passed argument rather than introducing a new format.

Reply:
You are right about some base64 encodings being valid hex, I'll fix that. 

But trimming the padding presents the same problem as supporting the padding to begin with. In certain circumstances there would be a possibility that we would use the wrong key. There is an old discussion on this on slack, if you want we can also discuss it there since it tends to be a long discussion.